Allow increased wait time for JS-induced redirect

oliverguenther opened the pull request:
Tries to select other element first after page redraw.

By selecting an entirely different element on the settings page
(e.g., the flash message), the button and radio elements are
being available afterwards and do not return a capybara cache error.

oliverguenther committed:
Allow increased wait time for JS-induced redirect

The same shared examples of `create_repository_spec` are repeatedly encountering timeouts
after a repository is created and the user is redirected to the settings page.

This allows these methods to have more wait time at their disposal.

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-repo-specs-increase-wait.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-repo-specs-increase-wait.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

ulferts commented on spec/features/repositories/create_repository_spec.rb:
```diff
@@ -152,7 +153,9 @@
-        expect(page).to have_selector('button[type="submit"]', text: I18n.t(:button_save))
+        expect(page).to have_selector('button[type="submit"]',
+                                      text: I18n.t(:button_save),
+                                      wait: 10)
```
Just a gut feeling, but I would still try to add an expectation for the flash message before I try to guess the right amount of wait time. 

I believe that selenium/capybara has problems here because the element you query for is there before and after the reloading (or an element with the same selector) and thus it might change while capybara is interacting with it.

When adding a safeguard in the form of an expectation on the flash message, an element that has not been there before, this should no longer be a problem.

oliverguenther committed:
Try to select other element first after page redraw.

By selecting an entirely different element on the settings page
(e.g., the flash message), the button and radio elements are
being available afterwards and do not return a capybara cache error.

NobodysNightmare commented:
Marking as WIP as you still seem to be trying out stuff.

In the current form you do not even modify any wait times ^^

oliverguenther commented:
I've updated the PR with the solution by ulferts. The wait times were the previous commit ;) Let me update the PR comment really quick

ulferts commented:
Cuke failure is due to code coverage not being uploaded correctly.
